[rpc] rename join_rpc to wait_all_workers in public api

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30052 [rpc] polish examples in docstrings and update docs to reflect correct use of
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**

Renames this API to wait_all_workers as discussed.

Differential Revision: [D18581466](https://our.internmc.facebook.com/intern/diff/D18581466/)

rohan-varma committed:
[rpc] rename join_rpc to wait_all_workers in public api

Renames this API to wait_all_workers as discussed.

Differential Revision: [D18581466](https://our.internmc.facebook.com/intern/diff/D18581466/)

[ghstack-poisoned]

rohan-varma commented on torch/distributed/rpc/api.py:
```diff
@@ -28,7 +28,7 @@ def wrapper(*args, **kwargs):
 
 
-def join_rpc():
+def wait_all_workers():
```
calling it `wait_all_workers` as per discussion, though for 1.4, it will probably still join all threads, which is more of a shutdown operation (meaning that user can't keep sending rpcs after this). maybe `join_all_workers` is more appropriate?

pritamdamania87 commented on torch/distributed/rpc/api.py:
```diff
@@ -28,7 +28,7 @@ def wrapper(*args, **kwargs):
-def join_rpc():
+def wait_all_workers():
     r"""
     Block until all local and remote RPC processes reach this method, process
```
Can we expand the docstring here to clearly indicate that this method should be used as part of termination and there is no guarantee that the RPC framework will work after this method returns. Would be also helpful to have an example here of how to just `wait_all_workers` and `shutdown` together.

rohan-varma committed:
Update on "[rpc] rename join_rpc to wait_all_workers in public api"

rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**

Renames this API to wait_all_workers as discussed.

Differential Revision: [D18581466](https://our.internmc.facebook.com/intern/diff/D18581466/)

[ghstack-poisoned]

rohan-varma commented:
@pritamdamania87 updated to include the changes to documentation, as well as adding a small example. Did not include `shutdown` since that PR has not landed.

rohan-varma committed:
Update on "[rpc] rename join_rpc to wait_all_workers in public api"

rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**

Renames this API to wait_all_workers as discussed.

Differential Revision: [D18581466](https://our.internmc.facebook.com/intern/diff/D18581466/)

[ghstack-poisoned]

mrshenli commented:
cc @zzzwen @mingzhe09088 This API change might break your code.

pritamdamania87 commented on torch/distributed/rpc/api.py:
```diff
@@ -28,26 +28,43 @@ def wrapper(*args, **kwargs):
+
+        On worker 0:
+        >>> import torch.distributed.rpc as rpc
+        >>> rpc.init_rpc("worker0", self_rank=0, world_size=2)
```
nit: rank instead of self_rank,

pritamdamania87 commented on torch/distributed/rpc/api.py:
```diff
@@ -28,26 +28,43 @@ def wrapper(*args, **kwargs):
-def join_rpc():
+def wait_all_workers():
     r"""
     Block until all local and remote RPC processes reach this method, process
```
Remove the part which says: "process all pending messages". This is true today for the ProcessGroup backend, but would not necessarily be true for other backends.

rohan-varma committed:
Update on "[rpc] rename join_rpc to wait_all_workers in public api"

rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**

Renames this API to wait_all_workers as discussed.

Differential Revision: [D18581466](https://our.internmc.facebook.com/intern/diff/D18581466/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] rename join_rpc to wait_all_workers in public api"

rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**

Renames this API to wait_all_workers as discussed.

Differential Revision: [D18581466](https://our.internmc.facebook.com/intern/diff/D18581466/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] rename join_rpc to wait_all_workers in public api"

rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**
rpc.init_rpc
* **#30050 [rpc] rename join_rpc to wait_all_workers in public api**

Renames this API to wait_all_workers as discussed.

Differential Revision: [D18581466](https://our.internmc.facebook.com/intern/diff/D18581466/)

[ghstack-poisoned]

rohan-varma commented:
CI errors are unrelated, landing this.
